### ISSUE TITLE
Entity sensor.luz_corredor_1_andar_energy_meter (<class 'custom_components.smartthings.sensor.SmartThingsSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('energy') it is using; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

### ISSUE THREAD

**Issue by rqmartins73** [veista/smartthings#56]
Good-afternon,
And sorry if this is already reported.

I just installed yesterday this Integration "by the book" and I'm getting this errors:

Entity sensor.luz_corredor_1_andar_energy_meter (<class 'custom_components.smartthings.sensor.SmartThingsSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('energy') it is using; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

Entity sensor.luz_escadas_energy_meter (<class 'custom_components.smartthings.sensor.SmartThingsSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('energy') it is using; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

Entity sensor.fibaro_switch_2_garagem_energy_meter (<class 'custom_components.smartthings.sensor.SmartThingsSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('energy') it is using; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

Entity sensor.frigorifico_cooler_temperature (<class 'custom_components.smartthings.sensor.SamsungOcfTemperatureSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('temperature') it is using; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

Entity sensor.fibaro_switch_garagem_energy_meter (<class 'custom_components.smartthings.sensor.SmartThingsSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('energy') it is using; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

Regards.
Thanks in advance.

**Comment by veista** [comment]
Hi,

The device does not report its unit of measurement. You can use the official integration for these devices.

**Comment by rqmartins73** [comment]
> Hi,
> 
> The device does not report its unit of measurement. You can use the official integration for these devices.

How can I have some devices on this Integration, and others in the Official?

**Comment by veista** [comment]
Yeah thats a good point. Im working on an updated integration which addresses most of the problems, including this type of problem.

**Comment by veista** [comment]
Yeah thats a good point. Im working on an updated integration which addresses most of the problems, including this type of problem.

**Comment by veista** [comment]
> > Hi,
> > 
> > The device does not report its unit of measurement. You can use the official integration for these devices.
> 
> How can I have some devices on this Integration, and others in the Official?
> 

If you want, you can do this by changing the domain in manifest.json to something else, also change it from const.py. after restart you should be able to run both simultaneously. Then you can separate the ocf devices to a different account or just cross disable the entities.

**Comment by rqmartins73** [comment]
> > > Hi,
> > > The device does not report its unit of measurement. You can use the official integration for these devices.
> > 
> > 
> > How can I have some devices on this Integration, and others in the Official?
> 
> If you want, you can do this by changing the domain in manifest.json to something else (for example smatthings_custom), also change it from const.py. after restart you should be able to run both simultaneously. Then you can separate the ocf devices to a different account or just cross disable the entities.

I'll wait for your fixes.
Thanks.

**Comment by rqmartins73** [comment]
For now I customized the entities with the kWh measure unit.
